sample_apps_reinvent2021/go/crud-ingestion-sample.go

```diff
@@ -0,0 +1,218 @@
+package main
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/timestreamquery"
+	"github.com/aws/aws-sdk-go/service/timestreamwrite"
+	"go_sample/utils"
+	"os"
+	"strconv"
+	"time"
+)
+
+/**
+  This code sample is to run the CRUD APIs and WriteRecords API in a logical order.
+*/
+
+func main() {
+
+	databaseName := flag.String("database_name", utils.DATABASE_NAME, "database name string")
+	tableName := flag.String("table_name", utils.TABLE_NAME, "table name string")
+	kmsKeyId := flag.String("kms_key_id", "", "kms key id for update database string")
+	region := flag.String("region", "us-east-1", "region")
+
+	flag.Parse()
+
+	// common set of dimensions used for ingestion
+	var dimensions []*timestreamwrite.Dimension
+	dimensions = append(dimensions, &timestreamwrite.Dimension{
+		Name:  aws.String("region"),
+		Value: aws.String("us-east-1"),
+	}, &timestreamwrite.Dimension{
+		Name:  aws.String("az"),
+		Value: aws.String("az1"),
+	}, &timestreamwrite.Dimension{
+		Name:  aws.String("hostname"),
+		Value: aws.String("host1"),
+	})
+
+	sess, err := utils.GetSession(*region)
+	utils.HandleError(err, "Failed to start a new session", true)
+
+	writeSvc := timestreamwrite.New(sess)
+	querySvc := timestreamquery.New(sess)
+	s3Svc := s3.New(sess, aws.NewConfig().WithRegion(*region))
+
+	reader := bufio.NewReader(os.Stdin)
+	fmt.Println("Creating a database, hit enter to continue")
+	reader.ReadString('\n')
```

Comment:
Why do we need user input after each step?

Reply:
I see. so this is similar to current application.   but I agree we do not need explicit customer input every time.